tests/xharness/TestTasks/TestTask.cs

```diff
@@ -10,70 +10,61 @@
+		public bool InProgress => (ExecutionResult & TestExecutingResult.InProgress) == TestExecutingResult.InProgress;
+		public bool Waiting => (ExecutionResult & TestExecutingResult.Waiting) == TestExecutingResult.Waiting;
+		public bool Finished => (ExecutionResult & TestExecutingResult.Finished) == TestExecutingResult.Finished;
 
-			if (RootDrive == null)
-				RootDrive = new DriveInfo ("/");
-			var afs = RootDrive.AvailableFreeSpace;
-			const long minSpaceRequirement = 1024 * 1024 * 1024; /* 1 GB */
-			if (afs < minSpaceRequirement) {
-				FailureMessage = $"Not enough space on the root drive '{RootDrive.Name}': {afs / (1024.0 * 1024):#.##} MB left of {minSpaceRequirement / (1024.0 * 1024):#.##} MB required";
-				ExecutionResult = TestExecutingResult.Failed;
+		public bool Building => (ExecutionResult & TestExecutingResult.Building) == TestExecutingResult.Building;
+		public bool Built => (ExecutionResult & TestExecutingResult.Built) == TestExecutingResult.Built;
+		public bool Running => (ExecutionResult & TestExecutingResult.Running) == TestExecutingResult.Running;
+
+		public bool BuildSucceeded => (ExecutionResult & TestExecutingResult.BuildSucceeded) == TestExecutingResult.BuildSucceeded;
+		public bool Succeeded => (ExecutionResult & TestExecutingResult.Succeeded) == TestExecutingResult.Succeeded;
+		public bool Failed => (ExecutionResult & TestExecutingResult.Failed) == TestExecutingResult.Failed;
+		public bool Ignored {
+			get => ExecutionResult == TestExecutingResult.Ignored;
+			set {
+				if (ExecutionResult != TestExecutingResult.NotStarted && ExecutionResult != TestExecutingResult.Ignored)
+					throw new InvalidOperationException ();
+				ExecutionResult = value ? TestExecutingResult.Ignored : TestExecutingResult.NotStarted;
 			}
-			return Task.CompletedTask;
 		}
+		public bool DeviceNotFound => ExecutionResult == TestExecutingResult.DeviceNotFound;
 
-		public void CloneTestProject (TestProject project)
-		{
-			// Don't build in the original project directory
-			// We can build multiple projects in parallel, and if some of those
-			// projects have the same project dependencies, then we may end up
-			// building the same (dependent) project simultaneously (and they can
-			// stomp on eachother).
-			// So we clone the project file to a separate directory and build there instead.
-			// This is done asynchronously to speed to the initial test load.
-			TestProject = project.Clone ();
-			InitialTask = TestProject.CreateCopyAsync ();
-		}
+		public bool Crashed => (ExecutionResult & TestExecutingResult.Crashed) == TestExecutingResult.Crashed;
+		public bool TimedOut => (ExecutionResult & TestExecutingResult.TimedOut) == TestExecutingResult.TimedOut;
+		public bool BuildFailure => (ExecutionResult & TestExecutingResult.BuildFailure) == TestExecutingResult.BuildFailure;
+		public bool HarnessException => (ExecutionResult & TestExecutingResult.HarnessException) == TestExecutingResult.HarnessException;
```

Comment:
A lot of code moved in this file, I have no idea if there are any actual changes or not.

Reply:
No changes in the code at all.